### PR TITLE
support `usm_ndarray` in `onedal.spmd`

### PR DESCRIPTION
**Pull request by samir-nasibli**
# Description

* adding usm_ndarray support for onedal module.
* wrapped functions
* updated SPMD examples with using dpctl tensors as input.

### CODE REVIEW

File: onedal/_device_offload.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+try:
+    from sklearnex._device_offload import (_get_global_queue,
+                                           _transfer_to_host,
+                                           _copy_to_usm)
+    _sklearnex_available = True
+except ImportError:
+    import logging
+    # TODO:
+    # update log message.
+    logging.warning('Device support is limited in daal4py patching. '
```
Comment (napetrov) [inline onedal/_device_offload.py:28]: should this warning be changed/removed?

File: onedal/_device_offload.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
+#===============================================================================
+# Copyright 2014 Intel Corporation
```
Comment (napetrov) [inline onedal/_device_offload.py:2]: if this new file we need 2023 year

**Review by KulikovNikita — APPROVED** [review]
Looks fine for me

**Review by ethanglaser — APPROVED** [review]
Looks great, all issues from yesterday are resolved. Examples run in my env.

**Review by napetrov — APPROVED** [review]
Let's wait for CI and merge

File: setup_sklearnex.py [hunk 3]
```diff
@@ -89,8 +89,13 @@
...
         'sklearnex.spmd',
-        'sklearnex.spmd.ensemble',
-        'sklearnex.spmd.linear_model']
+        'sklearnex.spmd.decomposition']
+    if ONEDAL_VERSION >= 20230100:
+        packages_with_tests += [
+            'sklearnex.spmd.basic_statistics',
+            'sklearnex.spmd.ensemble',
+            'sklearnex.spmd.linear_model',
+            'sklearnex.spmd.neighbors']
```
Comment (samir-nasibli) [inline setup_sklearnex.py:98]: Could be some master merge artefact. `sklearnex.spmd.ensemble` is not depend on ONEDAL_VERSION >= 20230100.
```suggestion
        'sklearnex.spmd.decomposition',
        'sklearnex.spmd.ensemble']
    if ONEDAL_VERSION >= 20230100:
        packages_with_tests += [
            'sklearnex.spmd.basic_statistics',
            'sklearnex.spmd.linear_model',
            'sklearnex.spmd.neighbors']
```

**Review by jmhatre-work — APPROVED** [review]

**Review by napetrov — APPROVED** [review]

**Pull request merged**